File: logica/lista2/exercise17.py
print("######### Verificação de Número Perfeito #########");

number = int(input("Digite um número: "));

sum = 0;


for i in range(1, number):
  if number % i == 0:
    sum += i;

if sum == number:
  print("O número é perfeito!");
else:
  print("O número não é perfeito!");

File: javascript/lista1/exercise17.js
console.log('########### Ordena lista de números ###########');

const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function sortList(list) {
    return list.sort((a, b) => a - b);
}

listNumber = [];

function recursiveAsyncReadLine() {
  rl.question('Digite um número ou -1 para sair: ', (number) => {
    
    if (number == -1) {
        const result = sortList(listNumber);
        console.log(`A lista ordenada é: ${result}`);
        return rl.close();
    } else {
        console.log(number);
        console.log(listNumber);
        listNumber.push(parseFloat(number));
        recursiveAsyncReadLine();
    }
  });
}

recursiveAsyncReadLine();

